src/Stepper/index.js

```diff
@@ -4,79 +4,71 @@ import * as React from "react";
 import StepperStateless from "./StepperStateless";
 import validateIncrement from "../utils/validateIncrement";
 import validateDecrement from "../utils/validateDecrement";
+import useStateWithCallback from "../hooks/useStateWithCallback";
 
-import type { Props, State } from "./index";
+import type { Props } from "./index";
 
-class Stepper extends React.PureComponent<Props, State> {
-  state = {
-    value: this.props.defaultValue || 0,
-  };
-
-  setValueAndInjectCallback = (value: number) => {
-    const { onChange } = this.props;
+const Stepper = ({ onChange, defaultValue, ...props }: Props) => {
+  const [value, setValue] = useStateWithCallback<number>(defaultValue || 0, val => {
```

Comment:
I noticed that we can enhance the `useStateWithCallback` a bit. It would great to use it only like this:
```js
const [value, setValue] = useStateWithCallback<number>(defaultValue, onChange);
```
And add the callback eval into the `useStateWithCallback` directly:
```js
if (value !== prevValue) {
  if(callback) callback(value);
}
```
Can you? 🙏 
https://github.com/kiwicom/orbit-components/blob/master/src/hooks/useStateWithCallback.js#-L11L14


Another note: because you passing function bind, the callback function inside the hook would be re-created upon every render of this component. Don't do this for memoized functions/components:
```js
val => {
  if(onChange) {
    onChange(val);
  }
}
```
Rather than, use `useCallback` separately. It's not necessary if you gonna extend the `useStateWithCallback` as I wrote above.